fdbrpc/TokenSign.cpp

```diff
@@ -164,3 +229,193 @@ TEST_CASE("/fdbrpc/TokenSign") {
 	printf("%d runs OK\n", numIters);
 	return Void();
 }
+
+namespace {
+
+std::string decode64(const std::string& val) {
+	using namespace boost::archive::iterators;
+	using It = transform_width<binary_from_base64<std::string::const_iterator>, 8, 6>;
+	return boost::algorithm::trim_right_copy_if(std::string(It(std::begin(val)), It(std::end(val))),
+	                                            [](char c) { return c == '\0'; });
+}
+
+std::string encode64(const std::string& val) {
+	using namespace boost::archive::iterators;
+	using It = base64_from_binary<transform_width<std::string::const_iterator, 6, 8>>;
+	auto tmp = std::string(It(std::begin(val)), It(std::end(val)));
+	return tmp.append((3 - val.size() % 3) % 3, '=');
+}
+
+template <class GenFun, class ValidateFun>
+void tokenVerifyBench(CryptAlgo algo, GenFun& gen, ValidateFun& validate) {
+	constexpr int numTokens = 10'000;
+	constexpr double testTime = 10;
+	Arena arena;
+	mkcert::KeyPairRef keyPair;
+	switch (algo) {
+	case CryptAlgo::ES256:
+		keyPair = mkcert::KeyPairRef::make(arena);
+		break;
+	case CryptAlgo::HMACSHA256: {
+		// the following is not secure, but good enough for testing
+		// we create a 64 byte key
+		auto k = new (arena) uint8_t[64];
+		static_assert(64 % sizeof(double) == 0);
+		for (int i = 0; i < 64; i += sizeof(double)) {
+			double d = nondeterministicRandom()->random01();
+			memcpy(k + i, &d, sizeof(double));
+		}
+		StringRef key(k, 64);
+		keyPair.privateKeyDer = key;
+		keyPair.publicKeyDer = key;
+		break;
+	}
+	}
+	std::vector<decltype(gen(algo, keyPair.privateKeyDer))> tokens;
+	tokens.reserve(numTokens);
+	fmt::print("Generating tokens");
+	for (int i = 0; i < numTokens; ++i) {
+		tokens.push_back(gen(algo, keyPair.privateKeyDer));
+		if (i % 1000 == 0) {
+			fmt::print(".");
+		}
+	}
+	fmt::print("\n");
+	auto startTime = timer();
+	auto printTime = startTime;
+	double currentTime = startTime;
+	uint64_t iterations = 0;
+	uint64_t lastPrintedIterations = 0;
+	while (startTime + testTime > currentTime) {
+		auto t = tokens[deterministicRandom()->randomInt(0, tokens.size())];
+		validate(algo, t, keyPair.publicKeyDer);
```

Comment:
With ASSERT() surrounding this, validate() for FlatBuffers seem to work, but it's returning false for JWT.